### PR TITLE
Upgrade trends and detail views to Bootstrap 5

### PR DESCRIPTION
**Pull request by uhafner**
Bootstrap 5 has been released and requires some changes in markup and JS. Additionally, (due to the migration of ECharts to Bootstrap 5) trend charts are now configurable using a dialog.

Upstream PR: jenkinsci/echarts-api-plugin#124

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: src/main/java/hudson/tasks/junit/History.java [hunk 1]
```diff
@@ -66,46 +69,60 @@ public boolean historyAvailable() {
         return testObject.getRun().getParent().getBuilds().size() > 1;
     }
 
-    @JavaScriptMethod
+    @Deprecated
```
Comment (timja) [inline src/main/java/hudson/tasks/junit/History.java:72]: just remove it, it's a web method only

File: src/main/java/hudson/tasks/junit/History.java [hunk 2]
```diff
@@ -66,46 +69,60 @@ public boolean historyAvailable() {
...
                 return new TestResultTrendChart().create(pluggableStorage.getTrendTestResultSummary());
             }
         }
 
-        return new TestResultTrendChart().createFromTestObject(createBuildHistory(testObject), new ChartModelConfiguration());
+        return new TestResultTrendChart().createFromTestObject(createBuildHistory(testObject), chartModelConfiguration);
     }
 
-    @JavaScriptMethod
+    @Deprecated
```
Comment (timja) [inline src/main/java/hudson/tasks/junit/History.java:96]: remove also

**Comment by uhafner** [comment]
Seems that HTMLUnit does not understand modern JS :-( Browser and IntelliJ do not complain. This is one of the reasons why I removed HTMLUnit tests from my plugins. They break very often due to incompatibilities (and are soooo slow).

```
EChartsJenkinsApi.prototype.readConfiguration = function (id) {
    const specific = echartsJenkinsApi.readFromLocalStorage(id);
    const common = echartsJenkinsApi.readFromLocalStorage(trendDefaultStorageId);

    return {
        ...specific,
        ...common
    };
}
```

**Comment by timja** [comment]
Everything in this plugin is inherited, probably from when it was split from core. 

I’ll try take a peek at this soon and see if I can find a way forward

**Review by jtnord — COMMENTED** [review]
I thought there was a trend toward getting rid of shared js libs not adding more  (using webpack and creating isolated bundles for each plugin).

If this is CSS are we going to have collisions where BS 4 and BS 5 are in the same page
@EstherAF

**Comment by timja** [comment]
I think only the history page uses bootstrap and that will be explicitly loading bootstrap 4

**Comment by timja** [comment]
Now failing for https://github.com/HtmlUnit/htmlunit/issues/232

**Comment by uhafner** [comment]
I have a testing course in the winter semester again. If it is helpful we could include the JUnit plugin as project and restructure it so that the ATH tests will run inside the plugin build as well. There is already a test at https://github.com/jenkinsci/acceptance-test-harness/blob/master/src/test/java/plugins/JUnitPluginTest.java that just needs to be moved (and maybe extended). 

As already mentioned above, I also think using HTMLUnit In general is not very helpful as it is just a (slow and buggy) simulation of a real browser. It works quite well with the static HTML pages of the old style Jenkins UI but not with modern pages that use a lot of other JS libraries. I think it make more sense if we would use unit and integration tests only to verify the Java models and not the browser rendering. Browser render should only be tested (if at all) using ATH.

**Comment by timja** [comment]
> I have a testing course in the winter semester again. If it is helpful we could include the JUnit plugin as project and restructure it so that the ATH tests will run inside the plugin build as well. There is already a test at [https://github.com/jenkinsci/acceptance-test-harness/blob/master/src/test/java/plugins/JUnitPluginTest.java](https://github.com/jenkinsci/acceptance-test-harness/blob/master/src/test/java/plugins/JUnitPluginTest.java?rgh-link-date=2021-08-09T07%3A13%3A54Z) that just needs to be moved (and maybe extended).

If you could do that it would be great!

Absolutely onboard with the conversion of these to ATH tests.

(In the meantime if rhino adds support for what we need we'll be able to ship this)

File: pom.xml [hunk 3]
```diff
@@ -48,16 +48,7 @@
...
-        <dependency>
-            <groupId>io.jenkins.plugins</groupId>
-            <artifactId>checks-api</artifactId>
-        </dependency>
-        <dependency>
-            <groupId>io.jenkins.plugins</groupId>
-            <artifactId>checks-api</artifactId>
-            <scope>test</scope>
-            <classifier>tests</classifier>
+            <version>5.1.2-6-rc504.fd62772631c3</version>
```
Comment (timja) [inline pom.xml:51]:
```suggestion
            <version>5.1.2-6</version>
```

**Review by timja — APPROVED** [review]
good enough with the limitations we have, I'm not sure if the tests are actually testing much now given the endpoints aren't being called with JS disabled but let's get this in so we can move forward in bom

**Comment by timja** [comment]
I've pushed origin branches:

https://github.com/jenkinsci/junit-plugin/tree/1.53.x
https://github.com/jenkinsci/junit-plugin/tree/1.54.x

I spent sometime trying to backport it but found it difficult, lots of pom issues.

Feel free to submit PRs to those branches, otherwise probably won't do? I believe you have found a workaround in bom

**Comment by basil** [comment]
I somehow managed to find a way to work around this for now, but …

> otherwise probably won't do

… in general, if you aren't willing to do backports, then I suggest not increasing the baseline so aggressively.

**Pull request merged**